test-interface/src/main/scala/scala/scalanative/testinterface/signalhandling/SignalConfig.scala

```diff
@@ -0,0 +1,59 @@
+package scala.scalanative.testinterface.signalhandling
+
+import scalanative.runtime.Platform._
+import scala.scalanative.libc.stdlib._
+import scala.scalanative.libc.signal._
+import scala.scalanative.unsafe._
+
+private[testinterface] object SignalConfig {
+  def setDefaultHandlers(): Unit = {
+
+    /* Default handler for signals like SIGSEGV etc.
+     * Since it needs to be able to handle segmentation faults, it has to
+     * exit the program on call, otherwise it will keep being called indefinetely.
+     * This and IO operations in handlers leading to unspecified behavior mean we
+     * can only communicate through exitcode values. In bash programs,
+     * exitcode > 128 signifies a fatal signal n, where n = exitcode - 128.
+     * This is the convention used here.
+     */
+    val defaultHandler = CFuncPtr1.fromScalaFunction { (sig: CInt) =>
+      exit(128 + sig)
```

Comment:
It turns out I misread a presentation - I thought that write() and read(), despite being async-signal-safe, could still lead to errors when used in signal handler, leading to my original comment about IO operations. That is of course not the case. Moreover, I discovered that all of the libunwind methods used in StackTrace_PrintStackTrace() in StackTrace.c are also async-signal-safe according to http://www.nongnu.org/libunwind/man/libunwind(3).html#section_5. This means that we should be able to print the stack trace, after rewriting aforementioned method to support write() instead of printf().

Reply:
Since I have one case where the stack unwind is the crash, it might be good to try and print the signal caught first and the stacktrace and then exit.